api/clients/v2/verification/cert_verifier.go

```diff
@@ -86,14 +94,35 @@ func (cv *CertVerifier) CheckDACert(
 
 	// TODO: determine if there's any merit in passing call options to impose better determinism and
 	// safety on the operation
-	result, err := certVerifierCaller.CheckDACert(
-		&bind.CallOpts{Context: ctx},
-		certBytes,
-	)
+
+	callMsgBytes, err := cv.v2VerifierBinding.TryPackCheckDACert(certBytes)
+	if err != nil {
+		return &CertVerifierInternalError{Msg: "pack checkDACert call", Err: err}
+	}
+
+	// Perform an eth_call with a gas limit of 20M or 2/3 of the current block gas limit.
+	// On-chain (e.g. in a rollup prover tx), budget additional gas since checkDACert adds
+	// to the overall execution path. The 2/3 factor is a heuristic and may be tuned further.
+	returnData, err := cv.ethClient.CallContract(ctx, ethereum.CallMsg{
+		To:   &certVerifierAddr,
+		Gas:  MaxGasPerCheckDACert,
+		Data: callMsgBytes,
+	}, nil)
+
+	if err != nil && IsEVMOutOfGasError(err) {
+		panic(fmt.Errorf("eth_call to checkDACert ran out of gas. "+
+			"This should not occur with a 2/3 block gas limit allocation, "+
+			"indicating a malformed certificate, potential exploit, or RPC issue: %w", err))
+	}
```

Comment:
don't panic this will return a 500 to user in proxy. We prob want to return a structured error to decide what happens with this (prob drop using a derivation pipeline error?)

Reply:
good question...